canonical-web-and-design/maas-ui
# feat(ui): move Add KVM form to header

## Done

- Moved "Add KVM" form to KVM list header
- Renamed `KVMActionForms` to `KVMHeaderForms`
- Moved `AddKVM` component files to `app/kvm/components/KVMHeaderForms` so all the header forms are in the same place (which is why `Files changed` is so high)
- Replaced `PodAction` enum with `KVMHeader` const
- Updated utils for determining header title

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the KVM list and click "Add KVM"
- Check that the Add KVM form shows, and "Add KVM" shows in the title
- Go to a KVM details page and check that the KVM forms still open correctly with action title
- Check that the VM forms still open correctly with action title

## Fixes

Fixes #3017
Fixes canonical-web-and-design/app-squad#242

## Screenshot
![Screenshot 2021-09-03 at 14-56-37 KVM bolla MAAS](https://user-images.githubusercontent.com/25733845/131954421-1d4a8d3f-ed82-4484-9c3e-e1ea3d0eba80.png)
